crates/ink/src/lib.rs

```diff
@@ -105,3 +105,24 @@ pub use ink_primitives::{
     H256,
     U256,
 };
+
+#[cfg(feature = "std")]
+#[doc(hidden)]
+pub use linkme;
+
+#[cfg(feature = "std")]
+use ink_metadata::EventSpec;
+
+/// Any event which derives `#[derive(ink::EventMetadata)]` and is used in the contract
+/// binary will have its implementation added to this distributed slice at linking time.
+#[cfg(feature = "std")]
+#[linkme::distributed_slice]
+#[linkme(crate = linkme)]
+pub static CONTRACT_EVENTS: [fn() -> EventSpec] = [..];
```

Comment:
The name change is mostly for vanity reasons, but it has the benefit of not requiring a coordinated update of the `contract-build` from `cargo-contract`.